internal/parser/scanner/scanner_test.go

```diff
@@ -1,15 +1,79 @@
 package scanner
 
 import (
+	"fmt"
+	"math/rand"
 	"testing"
+	"time"
 )
 
 func Test_hasNext(t *testing.T) {
-	t.SkipNow()
+	for _, k := range keywordSlice {
+		input := k
+		scanner := New([]rune(input))
+		for scanner.HasNext() {
+			scanner.Next()
```

Comment:
This does not test if `HasNext` actually returns false if the input ends. Maybe check the position of the scanner after `scanner.HasNext` returns `false`.